osu.Game/Screens/Play/SaveFailedScoreButton.cs

```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable disable
+
+using System;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Threading;
+using osu.Game.Scoring;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Online;
+using osuTK;
+
+namespace osu.Game.Screens.Play
+{
+    public class SaveFailedScoreButton : CompositeDrawable
+    {
+        public Func<Task<ScoreInfo>> SaveReplay;
+        private Task<ScoreInfo> saveReplayAsync;
+        private ScoreInfo score;
+
+        private ScheduledDelegate saveScoreDelegate;
+
+        protected readonly Bindable<ImportState> State = new Bindable<ImportState>();
+
+        private DownloadButton button;
+        private ShakeContainer shakeContainer;
+
+        public SaveFailedScoreButton(Func<Task<ScoreInfo>> sr)
+        {
+            Size = new Vector2(50, 30);
+            SaveReplay = sr;
+        }
+
+        [BackgroundDependencyLoader(true)]
+        private void load(OsuGame game)
+        {
+            InternalChild = shakeContainer = new ShakeContainer
+            {
+                RelativeSizeAxes = Axes.Both,
+                Child = button = new DownloadButton
+                {
+                    RelativeSizeAxes = Axes.Both,
+                }
+            };
+
+            button.Action = () =>
+            {
+                switch (State.Value)
+                {
+                    case ImportState.Imported:
+                        game?.PresentScore(score, ScorePresentType.Gameplay);
```

Comment:
Dunno about this. Import shouldn't be required to watch a scores. maybe okay for now but the UX is pretty shoddy.

Reply:
so it should delete it and make button disabled after imported?